plugins/Previous/src/Previous.ts

```diff
@@ -0,0 +1,45 @@
+import { Plugin, PluginCallbacks, State, self, PluginCallbacksOnSetArgument } from '@hookstate/core';
+
+const PluginID = Symbol('Previous');
+
+type PreviousInstanceType<S = any> = {
+  (): PluginCallbacks & {
+    getPrevious(): S
+  }
+  previous: S | undefined
+}
+
+const PreviousInstance: PreviousInstanceType = () => ({
+  onSet(data: PluginCallbacksOnSetArgument) {
+    PreviousInstance.previous = data.previous
```

Comment:
If you attach this plugin to 2 different states, one will override the previous state of the other, because you allow only one instance of the previous state to be remembered. PreviousInstance.previous is effectively a global constant.